### PR TITLE
refac: OperationCorrelationId refactoring

### PR DESCRIPTION
**Pull request by lasrinnil**
# This PR will:
- Make it clearer that we do not in fact set the `CorrelationId` prop on `ServiceBusMessage`. We set the OperationCorrelationId.
- Refactor the `CorrelationId` to more correctly be named `OperationCorrelationId` 

# Do note
- The extension methods that are written for `ServiceBusMessage` and `ServiceBusReceivedMessage`, Internally handle if the requested `key` is already written or not present in the `ApplicationProperties` bag. 

## References

* #690

### CODE REVIEW

File: source/dotnet/Services/Infrastructure/ServiceBus/MessageMetaDataConstants.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace Energinet.DataHub.Wholesale.Infrastructure.ServiceBus
 {
     public static class MessageMetaDataConstants
```
Comment (BjarkeMeier) [inline source/dotnet/Services/Infrastructure/ServiceBus/MessageMetaDataConstants.cs:17]: Perhaps we should add some code doc about the relation to ADR-008.

Comment (lasrinnil) [reply source/dotnet/Services/Infrastructure/ServiceBus/MessageMetaDataConstants.cs:21]: Good idea - I have updated the class with a summary with a quick description and a link.

File: source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Azure.Messaging.ServiceBus;
+
+namespace Energinet.DataHub.Wholesale.Infrastructure.ServiceBus;
+
+public static class ServiceBusMessageExtensions
```
Comment (BjarkeMeier) [inline source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageExtensions.cs:19]: Perhaps we should add some code doc about the relation to ADR-008.

File: source/dotnet/Services/IntegrationTests/TestCommon/ServiceBusReceivedMessageExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Azure.Messaging.ServiceBus;
+using Energinet.DataHub.Wholesale.Infrastructure.ServiceBus;
+
+namespace Energinet.DataHub.Wholesale.IntegrationTests.TestCommon
+{
+    public static class ServiceBusReceivedMessageExtensions
```
Comment (BjarkeMeier) [inline source/dotnet/Services/IntegrationTests/TestCommon/ServiceBusReceivedMessageExtensions.cs:20]: Why not just use the production code extension?

File: source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageFactory.cs [hunk 4]
```diff
@@ -42,17 +42,15 @@ public IEnumerable<ServiceBusMessage> Create(IEnumerable<byte[]> messages, strin
...
             Body = new BinaryData(body),
             Subject = messageType,
-            ApplicationProperties =
-            {
-                new KeyValuePair<string, object>(MessageMetaDataConstants.CorrelationId, correlationContextId),
-                new KeyValuePair<string, object>(MessageMetaDataConstants.MessageType, messageType),
-            },
         };
+        serviceBusMessage.SetOperationCorrelationId(operationCorrelationId);
+        serviceBusMessage.SetMessageType(messageType);
```
Comment (BjarkeMeier) [inline source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageFactory.cs:53]: Do we have tests ensuring that all required meta data has been set?

File: source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+{
+    public static void SetOperationCorrelationId(this ServiceBusMessage serviceBusMessage, string operationCorrelationId)
+    {
+        serviceBusMessage.ApplicationProperties.Add(MessageMetaDataConstants.OperationCorrelationId, operationCorrelationId);
```
Comment (BjarkeMeier) [inline source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageExtensions.cs:23]: What if the value has already been set? Or, if it's being set to a new different value?
Aren't these cases smell? Perhaps we should throw exceptions in those cases?

File: source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageFactory.cs (hunk 4 above)
Comment (lasrinnil) [reply source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageFactory.cs:53]: Well no. But we use the same factory for sending internal messages, would those messages have to be required to have the same meta data?

File: source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageExtensions.cs (hunk 2 above)
Comment (lasrinnil) [reply source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageExtensions.cs:19]: We have that under the constants, which is sufficient for me. What would you like to document about ADR-008 here?

Comment (BjarkeMeier) [reply source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageExtensions.cs:19]: I would copy the same documentation here. It's hard to tell from this class alone that its raison d'etre is ADR-008.

File: source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageExtensions.cs (hunk 5 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageExtensions.cs:23]: Hmm, yes, but I'm sorry I don't understand it 🙈

File: source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageFactory.cs (hunk 4 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageFactory.cs:53]: Should we consider different factories for domain vs integration events?
Probably not now. Just a reflection on the future...

File: source/dotnet/Services/IntegrationTests/TestCommon/ServiceBusReceivedMessageExtensions.cs (hunk 3 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/IntegrationTests/TestCommon/ServiceBusReceivedMessageExtensions.cs:20]: My bad. Thanks for explaining.

File: source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageExtensions.cs (hunk 5 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/Infrastructure/ServiceBus/ServiceBusMessageExtensions.cs:28]: Thx for code doc :-)

**Review by BjarkeMeier — APPROVED** [review]

**Pull request merged**